src/init.cpp

```diff
@@ -170,12 +171,26 @@ static fs::path GetPidFile(const ArgsManager& args)
 #else
         tfm::format(file, "%d\n", getpid());
 #endif
+        g_generated_pid = true;
         return true;
     } else {
         return InitError(strprintf(_("Unable to create the PID file '%s': %s"), fs::PathToString(GetPidFile(args)), SysErrorString(errno)));
     }
 }
 
+static void RemovePidFile(const ArgsManager& args)
+{
+    if (!g_generated_pid) return;
+    try {
+        if (!fs::remove(GetPidFile(args))) {
+            LogPrintf("Unable to remove PID file: File does not exist\n");
+        }
+    } catch (const fs::filesystem_error& e) {
+        LogPrintf("Unable to remove PID file: %s\n", fsbridge::get_filesystem_error_message(e));
+    }
```

Comment:
nit: can be simplified

```suggestion
    if (std::error_code error; !fs::remove(GetPidFile(args), error)) {
        std::string msg{error ? error.message() : "File does not exist"};
        LogPrintf("Unable to remove PID file: %s\n", msg);
    }
```

Reply:
I just like the brevity of it. I'm using `error.message()` and not `error.value()` so the user will see a user friendly error message. Could include the pid path in the log line if preferred, probably helpful to the user to know which file to change permissions on indeed.

For example, with this patch and `chmod a-w` on the datadir this will print:
> Unable to remove PID file: Permission denied

For a missing file, this will print
> Unable to remove PID file: File does not exist

Reply:
I've taken this in e56e8cdf9cda1712f4a415aa9af34ced3e78dfb2 and added the pid path to the log message.